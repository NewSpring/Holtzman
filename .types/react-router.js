declare module "react-router" {
  declare function Link(to: Object, className: string): Function;
  declare function withRouter(): Function;
}
