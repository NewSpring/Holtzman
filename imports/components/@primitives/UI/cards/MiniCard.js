
// @flow

import { Link } from "react-router";

type IMiniCard = {
  title: string,
  description: string,
  icon: string,
  category: string,
  link: string,
  image: string,
};

const MiniCard = ({
  title,
  description,
  icon,
  category,
  link,
  image,
}: IMiniCard) =>
  <Link to={link} className="plain">
    <div className="card">
      <div className={`card__item soft push-half-ends ${image ? "two-thirds" : "one-whole"}`} style={{ verticalAlign: "middle" }}>
        <h6 className="text-dark-primary capitalize">{ title }</h6>

        { description && <p className="text-dark-primary">{ description }</p> }

        <div className="display-inline-block">
          { icon && <span className={`${icon} text-dark-tertiary`} /> }
          { category && <h7 className="text-dark-tertiary soft-half-left">{ category }</h7> }
        </div>

      </div>
      <div
        className="locked-ends locked-right card__image one-third background--fill"
        style={{ verticalAlign: "middle", backgroundImage: `url('${image}')` }}
      />
    </div>
  </Link>
;

export default MiniCard;
