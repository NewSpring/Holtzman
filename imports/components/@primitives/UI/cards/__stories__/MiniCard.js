/* eslint-disable */
import { storiesOf } from '@storybook/react';
import {
  withKnobs,
  text,
  select,
  boolean,
} from '@storybook/addon-knobs';
import withReadme from "storybook-readme/with-readme";
import backgrounds from "react-storybook-addon-backgrounds";
import centered from "/.storybook/decorators/centered";
import defaultColors from "/.storybook/defaults";

import Readme from "./mini-card.md";
import MiniCard from "../MiniCard";

const story = storiesOf("Cards", module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addDecorator(backgrounds(defaultColors("light-primary", "light-secondary", "dark-primary", "dark-secondary")))
  ;

// We don't use description quite yet. As such, it's not really styled.
// So let's not pass it on this story.
// description={text("description", "MiniCard description")}

story
  .add("MiniCard", withReadme(Readme, () => {
    // set channel name options
    const channelOptions = {
      articles: "Articles",
      devotionals: "Devotionals",
      news: "News",
      sermons: "Sermons",
      stories: "Stories",
    };

    const content = {
      channelName: "articles",
      content: {
        images: [{
          fileLabel: "2:1",
          url: "http://www.adweek.com/files/blogs/jeff-goldblum-apartments-hed-2015.jpg",
        }],
      },
      title: "MiniCard Title",
    };

    // Channel Name
    content.channelName = select("type", channelOptions, "articles");

    // Image
    const defaultImageURL = "http://www.adweek.com/files/blogs/jeff-goldblum-apartments-hed-2015.jpg";
    content.content.images[0].url = text("image link", defaultImageURL);

    // Title
    const title = text("title", content.title);

    // Turn Off Image
    const hideImage = boolean("Hide Image?", false);
    if (hideImage) {
      content.content.images[0].url = "";
    }

    return (
      <div className={"floating"}>
        <div className={"grid__item text-left"} style={{ maxWidth: "480px" }}>
          <MiniCard
            title={title}
            content={content}
          />
        </div>
      </div>);
  }));
