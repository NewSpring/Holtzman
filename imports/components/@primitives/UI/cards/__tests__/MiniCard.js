
import { mount } from "enzyme"
import { mountToJson } from "enzyme-to-json";

import { Link } from "react-router";

import MiniCard from "../MiniCard";

describe("MiniCard", () => {
  const generateComponent = (additionalProps) =>
    <MiniCard {...additionalProps} />;

  it("should render with no props", () => {
    const component = mount(generateComponent());
    expect(component).toBeDefined();
    expect(mountToJson(component)).toMatchSnapshot();
  });

  it("should render title", () => {
    const component = mount(generateComponent({
      title: "Harambe",
    }));
    expect(component).toBeDefined();
    expect(component.find("h6").text()).toEqual("Harambe");
  });

  it("should render description", () => {
    const component = mount(generateComponent({
      description: "my description",
    }));
    expect(component).toBeDefined();
    expect(component.find("p")).toHaveLength(1);
    expect(component.find("p").text()).toEqual("my description");
  });

  it("should render with icon", () => {
    const component = mount(generateComponent({
      icon: "my-icon",
    }));
    expect(component).toBeDefined();
    expect(component.find("span")).toHaveLength(1);
    expect(component.find("span.my-icon")).toHaveLength(1);
  });

  it("should render with category", () => {
    const component = mount(generateComponent({
      category: "conspiracy",
    }));
    expect(component).toBeDefined();
    expect(component.find("h7")).toHaveLength(1);
    expect(component.find("h7").text()).toEqual("conspiracy");
  });

  it("should render with link", () => {
    const component = mount(generateComponent({
      link: "/wow",
    }));
    expect(component).toBeDefined();
    expect(component.find("Link")).toHaveLength(1);
    expect(component.find("Link").props().to).toEqual("/wow");
  });

  it("should render with image", () => {
    const component = mount(generateComponent({
      image: "/img",
    }));
    expect(component).toBeDefined();
    expect(component.find("div.card__item").props().className).toContain("two-thirds");
    expect(component.find("div.card__image").props().style.backgroundImage).toContain("/img");
  });

  it("should render full width with no image", () => {
    const component = mount(generateComponent());
    expect(component).toBeDefined();
    expect(component.find("div.card__item").props().className).toContain("one-whole");
  });
});
