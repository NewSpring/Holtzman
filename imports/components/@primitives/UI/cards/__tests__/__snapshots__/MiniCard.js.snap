exports[`MiniCard should render with no props 1`] = `
<MiniCard>
  <Link
    className="plain"
    onlyActiveOnIndex={false}
    style={Object {}}>
    <a
      className="plain"
      onClick={[Function]}
      style={Object {}}>
      <div
        className="card">
        <div
          className="card__item soft push-half-ends one-whole"
          style={
            Object {
              "verticalAlign": "middle",
            }
          }>
          <h6
            className="text-dark-primary capitalize push-half-top" />
        </div>
        <div
          className="locked-ends locked-right card__image one-third background--fill"
          style={
            Object {
              "backgroundImage": "url(\'undefined\')",
              "verticalAlign": "middle",
            }
          } />
      </div>
    </a>
  </Link>
</MiniCard>
`;
